### PR TITLE
Document `ACTOR_FLAG_UPDATE_DURING_OCARINA` and `PLAYER_STATE2_USING_OCARINA`

### PR DESCRIPTION
**Pull request by engineer124**
See https://github.com/zeldaret/oot/pull/2277

### CODE REVIEW

File: src/code/z_actor.c [hunk 1]
```diff
@@ -2541,8 +2541,8 @@ Actor* Actor_UpdateActor(UpdateActor_Params* params) {
     } else {
         if (!Object_IsLoaded(&play->objectCtx, actor->objectSlot)) {
             Actor_Kill(actor);
-        } else if ((params->requiredActorFlag && !(actor->flags & params->requiredActorFlag)) ||
-                   (((!params->requiredActorFlag) != 0) &&
+        } else if ((params->freezeExceptionFlag && !(actor->flags & params->freezeExceptionFlag)) ||
+                   (((!params->freezeExceptionFlag) != 0) &&
```
Comment (fig02) [inline src/code/z_actor.c:2545]: in oot we changed these conditions to use == 0 and != 0 because it isnt really a boolean check

Comment (fig02) [reply src/code/z_actor.c:2545]: `!params->freezeExceptionFlag) != 0` is extra weird actually lol, hope that isnt required to match or something

Comment (engineer124) [reply src/code/z_actor.c:2545]: The first `(params->freezeExceptionFlag != 0)` is fine, but yeah that second one does not want to match with different variants. That's the only thing that matches

Comment (fig02) [reply src/code/z_actor.c]: bizzare. wondering if some other change in the function could be made to allow that to work. Cause its weird that oot doesnt complain, even in retail versions. 
but, no idea what that could be

**Review by hensldm — APPROVED** [review]

**Review by AngheloAlf — APPROVED** [review]

**Pull request merged**